packages/babel-plugin-transform-nullish-coalescing-operator/test/fixtures/nullish-coalescing/logical-assignment/input.js

```diff
@@ -0,0 +1,4 @@
+let o;
+o ??= {};
+o.a ??= 1;
+o["b"] ??= 2;
```

Comment:
Can you add also a test case where `o` is undeclared (and thus, a property on `globalThis`)? We could use the https://babeljs.io/docs/assumptions#puregetters assumption for it.

Reply:
Can you add a test case for globalThis property with side effects?
```js
var access_a = 0;
Object.assign(globalThis, { get a() { return { count: access_a++ } } })

// should print 1
(a ?? {}).count
```

With the `pureGetters` assumption, we can safely assume that getting undeclared variable `a` does not introduce side effects so it is safe to generate `a !== void 0 && a !== null`.

Reply:
[repl](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=OYGw9gRghiAqAWBLAzgAgPztQd0QOwBMxsBuAKF0OIy1EhgRRKA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=Node-10.13&version=7.24.7&externalPlugins=&assumptions=%7B%22noDocumentAll%22%3Atrue%7D
)
This PR breaks this test, it seems like we have special treatment for `globalThis`, I'll take a look at it.